lib/utils/config.ts

```diff
@@ -165,3 +176,48 @@ export async function validateDevOptionAndWarn(
 		and exposes network ports such as 2375 that allows unencrypted access to balenaEngine.
 		Therefore, development mode should only be used in private, trusted local networks.`);
 }
+
+/**
+ * Chech whether the `--secureBoot` option of commands related to OS configuration
+ * such as `os configure` and `config generate` is compatible with a given
+ * device type and balenaOS version, and print a warning regarding the
+ * consequences of using that option.
+ */
+export async function validateSecureBootOptionAndWarn(
+	secureBoot?: boolean,
+	deviceManifest?: any,
```

Comment:
@alexgg  Do you think there is anyway we can make this strongly typed? 
If I understand the code correctly, we only care about the "yocto" and nested "machine" properties. Looking on where where this is called, I see that this should be of type `BalenaSdk.DeviceTypeJson.DeviceType` however, I don't see the "machine" property.  @thgreasi do you think we might be missing this yocto.machine property on the SDK typings https://github.com/balena-io/balena-sdk/blob/master/lib/types/device-type-json.ts#L30-L33 ?

Reply:
These should help get the typing in place for this one: https://github.com/balena-io/open-balena-api/pull/1351 and https://github.com/balena-io/balena-sdk/pull/1330

Reply:
right, not sure if we should expose `device-type.json` via typings, but I see both @thgreasi and yourself agree so I'm fine with it. I'll update the SDK version once the PR is merged and tagged.

Reply:
thanks, I updated the SDK and the code.